src/category_theory/limits/shapes/cartesian_closed.lean

```diff
@@ -321,7 +344,7 @@ def cartesian_closed_of_equiv (e : C ≌ D) [h : is_cartesian_closed C] : is_car
   { is_adj :=
     begin
       haveI q : exponentiable (e.inverse.obj X) := infer_instance,
-      have := q.is_adj,
+      have : is_left_adjoint (prod_functor.obj (e.inverse.obj X)) := q.is_adj,
```

Comment:
By the way I'm aware this proof is pretty horrible, feel free to improve/golf it!